### PR TITLE
Documentation developing loaders

### PR DESCRIPTION
**Pull request by zain-sohail**
Here I have added a guideline to start development.

We can add more comprehensive details to develop loaders here.
We can also improve the development workflow by addressing issues like adding pre commit hooks to the dev dependencies for example.

### CODE REVIEW

File: docs/misc/contributing.rst [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+
+
+2. **Install Python and Poetry:**
+   - Ensure you have Python 3.8, 3.9, or 3.10 and poetry installed.
+
+    .. code-block:: bash
+
+        pip install pipx
+        pipx install poetry
```
Comment (rettigl) [inline docs/misc/contributing.rst:38]: Is it necessary/advisable to use pipx here? Not sure I understand completely what this does.

File: docs/misc/contributing.rst [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    .. code-block:: bash
+
+        poetry install --dev
+
+
+Development Workflow
+=====================
+
+.. note::
+   This guide assumes that you have Python (version 3.8, 3.9, or 3.10) and poetry with dev dependencies installed on your machine.
```
Comment (rettigl) [inline docs/misc/contributing.rst:58]:
```suggestion
   This guide assumes that you have Python (version 3.8, 3.9, 3.10 or 3.11) and poetry with dev dependencies installed on your machine.
```

File: docs/misc/contributing.rst [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        git checkout -b feature-branch
+
+
+3. **Write Tests:** If your contribution introduces new features or fixes a bug, add tests to cover your changes.
+
+4. **Run Tests:** To ensure no funtionality is broken, run the tests:
+
+    .. code-block:: bash
+
+        pytest sed tests
```
Comment (rettigl) [inline docs/misc/contributing.rst:80]:
```suggestion
        pytest tests
```

**Review by rettigl — APPROVED** [review]
Mostly LGTM, apart from some small things.

File: docs/misc/contributing.rst (hunk 1 above)
Comment (zain-sohail) [reply docs/misc/contributing.rst:38]: Just an alternate to install poetry. It's the recommended version by poetry now. I tried it and it worked for me, much simpler than the alternates.

**Pull request merged**